contracts/zero-ex/contracts/src/features/native_orders/NativeOrdersInfo.sol

```diff
@@ -389,4 +393,22 @@ abstract contract NativeOrdersInfo is
             uint256(params.orderTakerAmount)
         ).safeDowncastToUint128();
     }
+
+    /// @dev checks if a given address is registered to sign on behalf of a maker address
+    /// @param maker the maker address encoded in an order (can be a contract)
+    /// @param signer the address that is providing a signature (an EOA)
+    function isValidOrderSigner(
+        address maker,
+        address signer
+    )
+        public
+        view
+        returns (bool isValid)
+    {
+        // returns false if it the mapping doesn't exist
+        return LibNativeOrdersStorage.getStorage()
```

Comment:
@mzhu25 is it inefficient to call getStorage() here if it's called in a function that already accesses storage?

Reply:
should be fine, `getStorage()` just computes the pointer, it doesn't do an `SLOAD`